### ISSUE TITLE
Is the Czech layout as intended, specifically the placement of "Z" a "Y"?

### ISSUE THREAD

**Issue by fuxoft** [8VIM/8VIM#369]
Is the Czech (Čeština) layout as intended? It has the same placement of "Z" and "Y" as the English layout but on standard Czech typewriter, Z and Y are switched because Z is more common in Czech. (Czech keyboards are in fact called "QWERTZ", not "QWERTY").

Otherwise, I am very happy to see 8pen resurrected.

**Comment by MaethorNaur** [comment]
Hi, yes currently all languages share the base English layout.
However, there is a project: https://github.com/kjoetom/8vim_keyboard_layout_calculator to generate a better layout.
I don't know how advanced this project is but it would be nice to migrate it to 8VIM and work on it. What do you think @kjoetom?

**Comment by fuxoft** [comment]
I have zero experience with smartphone development so I cannot be of much help as a programmer but I can give some pointers about the Czech language if necessary, e.g:

- Z is more common than Y.
- The letters Q and W are part of Czech alphabet but are not used in any Czech words, only in people's names and foreign words.
- The digraph "ch" (capitalized as "Ch") is a part of Czech alphabet (it goes after "H" and before "I") and is used very often, e.g. more often than the letter Y.
- The single quotation mark `'` is not used at all in Czech language. Never ever.
- Czech language has 42 letters, including letters with diacritics (e.g. "É", "Ě", "Ž", "Ý", "U", "Ú", "Ů"...). I know I can enter these letters using 8VIM but they are used in Czech text as often as other letters and it's a little illogical that e.g. "Ě" takes so much longer to enter than "E" when they are used approximately at the same rate in Czech text.

All of these nitpicks are valid only for people who write in Czech exclusively. On the other hand, the current Czech 8VIM layout is **very** useful when you often switch between Czech and English, which I personally do.

Let me know if you want more input.

Thanks again for what you've done.

**Comment by MaethorNaur** [comment]
I agree layout for languages other than English should be adapted accordingly. (I have the same issue for the French layout).

Also currently 8VIM doesn't really support digraphs (something we should work on). It will render only the first letter on the keyboard but insert the full digraph where you are typing.

Even if you don't have experience in mobile development, the layouts are just YAML files (https://github.com/8VIM/8VIM/wiki/Adding-a-new-language#language-file-layout). the Czech layout file is: https://github.com/8VIM/8VIM/blob/master/8vim/src/main/res/raw/cs.yaml
If you feel able to update it you're welcome

**Comment by fuxoft** [comment]
Is there a non-programmer way to test the usage of new layouts? I.e. I am able to edit YAML and copy something from/to my phone but I am not capable of compiling an APK.

**Comment by kjoetom** [comment]
> Is there a non-programmer way to test the usage of new layouts? I.e. I am able to edit YAML and copy something from/to my phone but I am not capable of compiling an APK.

Well, this is not necessary, because if you have a correct YAML file and it is stored somewhere on your mobile phone, you can load it directly from there via the settings of 8vim:

Settings --> "Select layout file from device".

and that's it.

:-)

**Comment by kjoetom** [comment]
If you want to easily create your own layout, with letters in a hidden layer, e.g. diacritics, maybe also with the possibility to write numbers directly on the xpad, you can use my 8vim-layout-generator ([here](https://github.com/8VIM/8VIM/discussions/295#discussion-5060023)). This is a Libreoffice spreadsheet where you just enter the desired localization of the letters graphically and save the results as a yaml file. Unfortunately it doesn't work with Excel, but with the free Libreoffice.

I updated the spreadsheet yesterday so that you can now use it to create your own layouts based on yaml.

kjoe

**Comment by fuxoft** [comment]
Thanks! LibreOffice is not a problem, all our computers only run Linux 😊

**Comment by MaethorNaur** [comment]
There's a tool that is used by the workflow: https://github.com/8VIM/json-yaml-validate. I could create a cli making it easier to use

**Comment by kjoetom** [comment]
> Hi, yes currently all languages share the base English layout. However, there is a project: https://github.com/kjoetom/8vim_keyboard_layout_calculator to generate a better layout. I don't know how advanced this project is but it would be nice to migrate it to 8VIM and work on it. What do you think @kjoetom?

I just noticed that I've been addressed here and apologize for not replying until now.

To be clear, this is not my project, I just forked it and made a small contribution. It's originally Glitchy-Tozier's project. 

See here: https://github.com/Glitchy-Tozier/8vim_keyboard_layout_calculator

Some time ago I used a different approach (based on Libreoffice) to optimize the layout for the German language. But Glitchy-Tozier's approach was always far superior to mine, much faster and more reliable ;-)
